app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt

```diff
@@ -45,323 +39,454 @@ import androidx.compose.ui.graphics.Color
+            }
+
+        // Display the count
+        Text(
+            text = servingsCount.toString(),
+            style = MaterialTheme.typography.titleMedium,
+            color = MaterialTheme.colorScheme.background,
+            textAlign = TextAlign.Center,
+            modifier = Modifier.testTag("numberServings").width(OVERVIEW_COUNTER_TEXT_SIZE.dp))
+
+        // + button
+        Button(
+            onClick = {
+              if (servingsCount < OVERVIEW_MAX_COUNTER_VALUE) onCounterChange(servingsCount + 1)
+            },
+            colors =
+                ButtonDefaults.buttonColors(
+                    containerColor = MaterialTheme.colorScheme.onBackground,
+                    contentColor = MaterialTheme.colorScheme.background),
+            modifier = Modifier.size(COUNTER_MIN_MAX_SIZE.dp).testTag("addServings"),
+            contentPadding = PaddingValues(OVERVIEW_RECIPE_COUNTER_PADDING.dp)) {
               Text(
-                  servingsCountVar.toString(),
-                  fontSize = 12.sp,
-                  color = Color.Black,
-                  modifier = Modifier.testTag("numberServings"))
-              Button(
-                  onClick = { changeServingsCount(++servingsCountVar) },
-                  colors = ButtonColors(goldenBronze, Color.Black, goldenBronze, Color.Black),
-                  modifier = Modifier.size(20.dp).testTag("addServings"),
-                  contentPadding = PaddingValues()) {
-                    Text("+", fontSize = 12.sp)
-                  }
+                  stringResource(R.string.counter_max),
+                  style = MaterialTheme.typography.titleMedium,
+                  color = MaterialTheme.colorScheme.background)
             }
       }
-  Spacer(modifier = Modifier.height(7.dp))
-  IngredientsList(currentRecipe)
 }
 
-/** Display of the list of ingredients */
+/**
+ * Extracts the first integer from the measurement string, doubles it and replaces it in the string
+ *
+ * @param measurement: The measurement string
+ * @param servingsCount: The number of servings
+ * @return The modified string
+ */
+private fun extractAndDoubleFirstInt(measurement: String, servingsCount: Int): String {
+  val regex = Regex("""\d+""")
+  val match = regex.find(measurement)
+  return if (match != null) {
+    val originalNumber = match.value.toInt()
+    val doubledNumber = originalNumber * servingsCount
+    measurement.replaceFirst(match.value, doubledNumber.toString())
+  } else {
+    measurement
+  }
+}
+
+/**
+ * Display of the ingredients
+ *
+ * @param currentRecipe: The recipe to display
+ * @param servingsCount: The number of servings
+ */
 @Composable
-private fun IngredientsList(currentRecipe: Recipe?) {
-  Column(modifier = Modifier.testTag("ingredientsView")) {
-    currentRecipe?.ingredientsAndMeasurements?.forEach { (ingredient, measurement) ->
+private fun IngredientsList(currentRecipe: Recipe, servingsCount: Int) {
```

Comment:
1. **Dynamic Measurement Handling**:
   - Nice use of `extractAndDoubleFirstInt` for dynamically scaling measurements based on servings! It keeps things responsive to the user’s input.

2. **Checkbox State**:
   - Using `remember` for `ticked` is great, but consider moving the state management outside of the loop if possible (e.g., in the `ViewModel`), especially if the ingredient list grows—it’ll prevent the state from resetting on recompositions.

3. **Accessibility for Ingredient Text**:
   - Consider adding a `contentDescription` to the ingredient `Text` to help screen readers interpret it better.

   **Quick Example**:
   ```kotlin
   Text(
       text = "$ingredient: $modifiedMeasurement",
       modifier = Modifier
           .testTag("ingredient$ingredient")
           .semantics { contentDescription = "$ingredient, $modifiedMeasurement" }
   )
   ```

Reply:
1 Thank you!
2. I can't move the state outside right now, as I need a separate `remember` for each tick. Moving it outside wouldn’t be helpful at this stage. It may make sense if we need to manage the fridge later on, but we can revisit it then.
3. Done!